pkgs/by-name/sn/snpguest/package.nix

```
@@ -0,0 +1,44 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  openssl,
+  stdenv,
+  darwin,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "snpguest";
+  version = "0.6.0";
+
+  src = fetchFromGitHub {
+    owner = "virtee";
+    repo = "snpguest";
+    rev = "v${version}";
+    hash = "sha256-9TchRaZPQKAsncs+mlHvzeie9IIVZeea/LfBLXOLuNg=";
+  };
+
+  cargoHash = "sha256-1UX5GiwH38W+IgZO+0EA3M86iWMylM8fgr48DRD187A=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+
+  env = {
+    OPENSSL_NO_VENDOR = true;
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "CLI tool for interacting with SEV-SNP guest environment";
+    homepage = "https://github.com/virtee/snpguest";
+    changelog = "https://github.com/virtee/snpguest/releases/tag/v${version}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ katexochen ];
+    mainProgram = "snpguest";
+    platform = "x86_64-linux";
```

```suggestion
    platforms = [ "x86_64-linux" ];
```

platform [isn't](https://gist.github.com/GrahamcOfBorg/fda86ff4f3398d2e8d437755ae2fcdc8) a thing.